carbon-design-system/carbon-addons-iot-react
# feat(table): Manage view modal

Closes #1308

**Summary**
A highly customisable modal built for managing table views. The component is built as component outside the actual table but only make sense in the context of table view management.

Default:
![image](https://user-images.githubusercontent.com/5167091/91210869-23899900-e70e-11ea-9803-d72f4a710a40.png)

Heavily customised:
![image](https://user-images.githubusercontent.com/5167091/91210890-2be1d400-e70e-11ea-8899-f6aab1f5a8c4.png)


**Change List (commits, features, bugs, etc)**
1. Created component TableManageViewsModal that lets a user list, search, filter and take actions on views. The state needed for search, pagination etc is handled by the application via callbacks.
2. Created a TableManageViewsList component as a subcomponent of the TableManageViewsModal that in turn uses the List compoinent.
3. Created 2 stories as a sub section of the table stories. One that demonstrates and with info explains the full usage and the other how to customise the modal. 
4. Wrote unit tests

**Acceptance Test (how to verify the PR)**
* Go to [/story/watson-iot-table-tablemanageviewsmodal--with-callbacks-implemented](https://deploy-preview-1537--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table-tablemanageviewsmodal--with-callbacks-implemented)
  * Test the knobs 
  * Test that all the functionality is working according to spec
  * Verify the info text
* Go to [story/watson-iot-table-tablemanageviewsmodal--with-custom-row-actions-custom-rendering-and-no-pagination](https://deploy-preview-1537--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table-tablemanageviewsmodal--with-custom-row-actions-custom-rendering-and-no-pagination)
  * Test that it looks ok and that the actions are working as storybook actions
  * Verify the info text
